File: StackArrays/stackarray.py
class Stack:
    MAX = 1000

    def __init__(self):
        self.top = -1
        self.arr = [0] * self.MAX  # Initialize stack with a fixed size

    def push(self, x):
        if self.top >= (self.MAX - 1):
            print("Stack Overflow")
            return False
        else:
            self.top += 1
            self.arr[self.top] = x
            print(f"{x} pushed into stack")
            return True

    def pop(self):
        if self.top < 0:
            print("Stack Underflow")
            return 0
        else:
            x = self.arr[self.top]
            self.top -= 1
            return x

    def peek(self):
        if self.top < 0:
            print("Stack is Empty")
            return 0
        else:
            return self.arr[self.top]

    def is_empty(self):
        return self.top < 0


if __name__ == "__main__":
    s = Stack()
    s.push(10)
    s.push(20)
    s.push(30)
    print(f"{s.pop()} popped from stack")
    print(f"Top element is: {s.peek()}")
    print(f"Stack empty: {s.is_empty()}")

File: StackArrays/stackarray.java
class Stack {
    private static final int MAX = 1000;
    private int top;
    private int[] arr;

    public Stack() {
        this.top = -1;
        this.arr = new int[MAX]; // Initialize stack with a fixed size
    }

    public boolean push(int x) {
        if (top >= (MAX - 1)) {
            System.out.println("Stack Overflow");
            return false;
        } else {
            arr[++top] = x;
            System.out.println(x + " pushed into stack");
            return true;
        }
    }

    public int pop() {
        if (top < 0) {
            System.out.println("Stack Underflow");
            return 0;
        } else {
            return arr[top--];
        }
    }

    public int peek() {
        if (top < 0) {
            System.out.println("Stack is Empty");
            return 0;
        } else {
            return arr[top];
        }
    }

    public boolean isEmpty() {
        return top < 0;
    }

    public static void main(String[] args) {
        Stack s = new Stack();
        s.push(10);
        s.push(20);
        s.push(30);
        System.out.println(s.pop() + " popped from stack");
        System.out.println("Top element is: " + s.peek());
        System.out.println("Stack empty: " + s.isEmpty());
    }
}
